### PR TITLE
Add dry-run/monitoring-only mode for TxThrottler

### PR DESCRIPTION
**Pull request by ejortegau**
## Description

This change adds the ability to run the transaction throttler in dry-run/ monitoring-only mode. This means that it will only emit metrics related to whether throttling would or would not have take place, but will not actually throttle. This can be useful when deploying, to get a sense of how often throttling can happen without actually having it happen.

## Related Issue(s)

https://github.com/vitessio/vitess/issues/13603

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

## Deployment Notes

N/A

### CODE REVIEW

**Comment by deepthi** [comment]
Docs:
- This seems important enough to include in release notes
- Website [program reference](https://github.com/vitessio/website/blob/prod/content/en/docs/18.0/reference/programs/vttablet.md) should be updated to include the new flag. I don't think we auto-generate those yet, so it needs to be done by hand.

**Comment by ejortegau** [comment]
> Docs:
> 
> * This seems important enough to include in release notes
> * Website [program reference](https://github.com/vitessio/website/blob/prod/content/en/docs/18.0/reference/programs/vttablet.md) should be updated to include the new flag. I don't think we auto-generate those yet, so it needs to be done by hand.

I have added info to the release notes & filed a PR for the website.

File: go/flags/endtoend/vttablet.txt [hunk 1]
```diff
@@ -347,6 +347,7 @@ Usage of vttablet:
       --twopc_enable                                                     if the flag is on, 2pc is enabled. Other 2pc flags must be supplied.
       --tx-throttler-config string                                       Synonym to -tx_throttler_config (default "target_replication_lag_sec:2 max_replication_lag_sec:10 initial_rate:100 max_increase:1 emergency_decrease:0.5 min_duration_between_increases_sec:40 max_duration_between_increases_sec:62 min_duration_between_decreases_sec:20 spread_backlog_across_sec:20 age_bad_rate_after_sec:180 bad_rate_increase:0.1 max_rate_approach_threshold:0.9")
       --tx-throttler-default-priority int                                Default priority assigned to queries that lack priority information (default 100)
+      --tx-throttler-dry-run                                             If present, the TxThrottler only records metrics about requests received and throttled, but does not actually throttle any requests.
```
Comment (timvaillancourt) [inline go/flags/endtoend/vttablet.txt:350]: One suggestion to be consistent with other `--help` flags:
```suggestion
      --tx-throttler-dry-run                                             If present, the transaction throttler only records metrics about requests received and throttled, but does not actually throttle any requests.
```

File: go/vt/vttablet/tabletserver/tabletenv/config.go [hunk 2]
```diff
@@ -185,6 +185,7 @@ func registerTabletEnvFlags(fs *pflag.FlagSet) {
 	flagutil.DualFormatStringListVar(fs, &currentConfig.TxThrottlerHealthCheckCells, "tx_throttler_healthcheck_cells", defaultConfig.TxThrottlerHealthCheckCells, "A comma-separated list of cells. Only tabletservers running in these cells will be monitored for replication lag by the transaction throttler.")
 	fs.IntVar(&currentConfig.TxThrottlerDefaultPriority, "tx-throttler-default-priority", defaultConfig.TxThrottlerDefaultPriority, "Default priority assigned to queries that lack priority information")
 	fs.Var(currentConfig.TxThrottlerTabletTypes, "tx-throttler-tablet-types", "A comma-separated list of tablet types. Only tablets of this type are monitored for replication lag by the transaction throttler. Supported types are replica and/or rdonly.")
+	fs.BoolVar(&currentConfig.TxThrottlerDryRun, "tx-throttler-dry-run", defaultConfig.TxThrottlerDryRun, "If present, the TxThrottler only records metrics about requests received and throttled, but does not actually throttle any requests.")
```
Comment (timvaillancourt) [inline go/vt/vttablet/tabletserver/tabletenv/config.go:188]: Another suggestion:
```suggestion
	fs.BoolVar(&currentConfig.TxThrottlerDryRun, "tx-throttler-dry-run", defaultConfig.TxThrottlerDryRun, "If present, the transaction throttler only records metrics about requests received and throttled, but does not actually throttle any requests.")
```

File: go/flags/endtoend/vttablet.txt (hunk 1 above)
Comment (ejortegau) [reply go/flags/endtoend/vttablet.txt]: Fixed

File: go/vt/vttablet/tabletserver/tabletenv/config.go (hunk 2 above)
Comment (ejortegau) [reply go/vt/vttablet/tabletserver/tabletenv/config.go]: Fixed

**Review by timvaillancourt — APPROVED** [review]

**Review by shlomi-noach — APPROVED** [review]

File: changelog/18.0/18.0.0/summary.md [hunk 3]
```diff
@@ -101,6 +101,14 @@ Vitess upgrade process from an earlier version if you need to use such a workflo
 Any MoveTables or Migrate workflow that moves a sequence table should only be run after all vitess components have been
 upgraded, and no upgrade should be done while such a workflow is in progress.
 
+#### <a id="vttablet-tx-throttler-dry-run"/>New Dry-run/monitoring-only mode for the transaction throttler
+
+A new CLI flag `--tx-throttler-dry-run` to set the Transaction Throttler to monitoring-only/dry-run mode has been added.
+If the transaction throttler is enabled with `--enable-tx-throttler` and the new dry-run flag is also specified, the
+tablet will not actually throttle any transactions; however, it will increase the counters for transactions throttled
+(`vttablet_transaction_throttler_throttled`). This allows to deploy the transaction throttler in production and gain
```
Comment (deepthi) [inline changelog/18.0/18.0.0/summary.md:109]:
```suggestion
(`vttablet_transaction_throttler_throttled`). This allows us to deploy the transaction throttler in production and gain
```

Comment (deepthi) [reply changelog/18.0/18.0.0/summary.md:109]: Or `users`. But that verb definitely needs an object.

**Review by deepthi — APPROVED** [review]
If you can fix the one nit on release notes, we can merge after that.

File: changelog/18.0/18.0.0/summary.md (hunk 3 above)
Comment (ejortegau) [reply changelog/18.0/18.0.0/summary.md]: Fixed.

**Pull request merged**